esp-hal/src/timer/timg.rs

```diff
@@ -303,6 +303,54 @@ where
 {
     /// Construct a new instance of [`TimerGroup`] in asynchronous mode
     pub fn new_async(_timer_group: impl Peripheral<P = T> + 'd) -> Self {
+        use crate::timer::timg::asynch::timg0_timer0_handler;
+        #[cfg(timg_timer1)]
+        use crate::timer::timg::asynch::timg0_timer1_handler;
+        #[cfg(timg1)]
+        use crate::timer::timg::asynch::timg1_timer0_handler;
+        #[cfg(all(timg1, timg_timer1))]
+        use crate::timer::timg::asynch::timg1_timer1_handler;
+
+        match T::id() {
+            0 => unsafe {
+                interrupt::bind_interrupt(Interrupt::TG0_T0_LEVEL, timg0_timer0_handler.handler());
+                interrupt::enable(Interrupt::TG0_T0_LEVEL, timg0_timer0_handler.priority())
+                    .unwrap();
+
+                #[cfg(timg_timer1)]
+                {
+                    interrupt::bind_interrupt(
+                        Interrupt::TG0_T1_LEVEL,
+                        timg0_timer1_handler.handler(),
+                    );
+                    interrupt::enable(Interrupt::TG0_T1_LEVEL, timg0_timer1_handler.priority())
+                        .unwrap();
+                }
+            },
+            #[cfg(any(timg1, timg_timer1))]
```

Comment:
```suggestion
            #[cfg(timg1)]
```

Reply:
Thanks! Because I had to touch the code anyway, I applied your suggestions there.